### PR TITLE
chore: IP rate limit WAF rule for Core

### PR DESCRIPTION
**Pull request by chejimmy**
# Description

Added a WAF rule to rate-limit traffic by IP for core service. The rate is at 3,000 over 5 minutes matching the NestJS rule ([code](https://github.com/awslabs/iot-application/blob/main/apps/core/src/app.module.ts#L19)). This addition protects the core service against DoS at a reasonable TPS.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- deployed and verified the rule
> Rate-limiting criteria
Request aggregation
Source IP address
Rate limit
3,000

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Legal

This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

**Review by dpportet — APPROVED** [review]
Looks good. Is there an easy way we can test this to make sure it works? Some script that hits the dashboard endpoint at over 10 TPS?

File: cdk/lib/core/core-service-web-acl.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      },
+      rules: [
+        {
+          action: { block: {} },
+          name: 'IoTAppCoreServiceRateLimit',
+          priority: 0,
+          statement: {
+            rateBasedStatement: {
+              aggregateKeyType: 'IP',
+              limit: 3000, // 10 TPS over 5 minutes
```
Comment (tracy-french) [inline cdk/lib/core/core-service-web-acl.ts:26]: How does someone determine what the limit should be?

**Review by tracy-french — APPROVED** [review]
Thanks @chejimmy ! This is a great add!

File: cdk/lib/core/core-service-web-acl.ts (hunk 1 above)
Comment (chejimmy) [reply cdk/lib/core/core-service-web-acl.ts:26]: > Looks good. Is there an easy way we can test this to make sure it works? Some script that hits the dashboard endpoint at over 10 TPS?

> How does someone determine what the limit should be?

We will be performing load testing as part of our operational readiness review
The artifact can help us and customers to determine the limit 👍

**Pull request merged**